numpy/_core/src/multiarray/item_selection.c

```diff
@@ -3085,43 +3112,83 @@ PyArray_Sort(PyArrayObject *op, int axis, NPY_SORTKIND flags)
+        PyArray_Descr *descr = PyArray_DESCR(op);
+        PyArray_DTypeMeta *dt = NPY_DTYPE(descr);
+
+        PyArray_DTypeMeta *dtypes[2] = {dt, dt};
+        PyArray_Descr *given_descrs[2] = {descr, descr};
+        PyArray_Descr *loop_descrs[2];
+        npy_intp view_offset = 0;
+        
+        if (sort_method->resolve_descriptors(
+            sort_method, dtypes, given_descrs, loop_descrs, &view_offset) < 0) {
+            PyErr_SetString(PyExc_RuntimeError,
+                            "unable to resolve descriptors for sort");
             return -1;
         }
+        
+        PyArrayMethod_SortParameters sort_params = {
+            .flags = flags,
+        };
+        context.parameters = &sort_params;
+        context.descriptors = loop_descrs;
+
+        // Arrays are always contiguous for sorting
+        npy_intp stride = PyArray_ITEMSIZE(op);
+        npy_intp strides[2] = {stride, stride};
+        NPY_ARRAYMETHOD_FLAGS method_flags = 0;
+
+        if (sort_method->get_strided_loop(
+            &context, 1, 0, strides, &strided_loop, &auxdata, &method_flags) < 0) {
+            PyErr_SetString(PyExc_RuntimeError,
+                            "unable to get strided loop for sort");
+            return -1;
+        }
+    }
+    else {
+        sort_table = PyDataType_GetArrFuncs(PyArray_DESCR(op))->sort;
         switch (flags) {
             case NPY_SORT_DEFAULT:
-                sort = generic_sort_table[NPY_QUICKSORT];
+                sort = sort_table[NPY_QUICKSORT];
                 break;
             case NPY_SORT_STABLE:
-                sort = generic_sort_table[NPY_STABLESORT];
+                sort = sort_table[NPY_STABLESORT];
                 break;
             default:
                 break;
         }
-    }
 
-    if (sort == NULL) {
-        PyErr_SetString(PyExc_TypeError,
-                        "no current sort function meets the requirements");
-        return -1;
+        // Look for appropriate generic function if no type specific version
+        if (sort == NULL) {
+            if (!PyDataType_GetArrFuncs(PyArray_DESCR(op))->compare) {
+                PyErr_SetString(PyExc_TypeError,
+                                "type does not have compare function");
+                return -1;
+            }
+            switch (flags) {
+                case NPY_SORT_DEFAULT:
+                    sort = generic_sort_table[NPY_QUICKSORT];
+                    break;
+                case NPY_SORT_STABLE:
+                    sort = generic_sort_table[NPY_STABLESORT];
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        if (sort == NULL) {
+            PyErr_SetString(PyExc_TypeError,
+                            "no current sort function meets the requirements");
+            return -1;
+        }
     }
-    return _new_sortlike(op, axis, sort, NULL, NULL, 0);
+
+    return _new_sortlike(op, axis, sort, strided_loop,
```

Comment:
So, I'd change this to
```
ret = _new_sortlike(...);
NPY_AUXDATA_FREE(aux_data);
Py_DECREF(context->decriptors[0]);
Py_DECREF(context->decriptors[1]);
// Anything else?
return ret;
```

Reply:
Conditional on the `strided_loop` path being taken, I suppose? That is done, thanks!